theano/gpuarray/opt.py

```diff
@@ -845,10 +848,14 @@ def local_gpu_pdbbreakpoint_op(node):
 def local_gpua_lazy_ifelse(op, context_name, inputs):
     if op.gpu:
         return
+    if isinstance(inputs[0].type, GpuArrayType):
+        return
     c = inputs[0]
     inps = []
     for v in inputs[1:]:
-        if isinstance(v.type, (tensor.TensorType, GpuArrayType)):
+        if isinstance(v.type, GpuArrayType):
+            return
+        elif isinstance(v.type, tensor.TensorType):
```

Comment:
Just for the sake of clarity: I think what @nouiz said is no longer correct. `inputs` are the equivalent of the inputs in the `mapping`, so if they are on GPU, all it indicates is that the previous Op has been transferred to the GPU, it says nothing about the current one.
If you check the outputs, then that is another story.